libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -522,15 +526,33 @@ private void initializeMapLayerInteractor(MapboxMap mapboxMap) {
   }
 
   private void initializeStreetsSource(MapboxMap mapboxMap) {
-    VectorSource streetSource = new VectorSource(STREETS_SOURCE_ID, MAPBOX_STREETS_V7);
-    mapboxMap.getStyle().addSource(streetSource);
-    LineLayer streetsLayer = new LineLayer(STREETS_LAYER_ID, STREETS_SOURCE_ID)
-      .withProperties(
-        lineWidth(DEFAULT_WIDTH),
-        lineColor(Color.WHITE)
-      )
-      .withSourceLayer(ROAD_LABEL);
-    mapboxMap.getStyle().addLayerAt(streetsLayer, LAST_INDEX);
+    List<Source> sources = mapboxMap.getStyle().getSources();
+    Source sourceV7 = findSourceById(sources, MAPBOX_STREETS_V7_URL);
+    Source sourceV8 = findSourceById(sources, MAPBOX_STREETS_V8_URL);
+
+    if (sourceV7 != null) {
+      layerInteractor.addStreetsLayer(sourceV7.getId(), STREETS_V7_ROAD_LABEL);
+    } else if (sourceV8 != null) {
+      layerInteractor.addStreetsLayer(sourceV8.getId(), STREETS_V8_ROAD_LABEL);
+    } else {
+      VectorSource streetSource = new VectorSource(STREETS_SOURCE_ID, MAPBOX_STREETS_V8_URL);
+      mapboxMap.getStyle().addSource(streetSource);
+      layerInteractor.addStreetsLayer(STREETS_SOURCE_ID, STREETS_V8_ROAD_LABEL);
+    }
+  }
+
+  @Nullable
+  private Source findSourceById(List<Source> sources, String streetsUrl) {
```

Comment:
This method name threw me for a momentary loop: it’s actually finding the source by URL (specifically, by a tileset ID wrapped in a URL).